### DISCUSSION TITLE
How to create grid with max-width first column

### DISCUSSION THREAD

**Discussion by somguyth** [chakra-ui/chakra-ui#4436] [Q&A] [answered]
I've successfully used `Grid` and `GridItem` described [here](https://chakra-ui.com/docs/layout/grid) to create something like this:

![image](https://user-images.githubusercontent.com/37128022/126707012-c5f1105a-febe-4520-8198-2e32c2e4c575.png)

However, I want to make it so that the first column in the grid has a maxWidth of, say, 50px, no matter how wide the browser window gets. Is there a way I can achieve that result? I don't see any such props in `GridItem` to specify a max-width (perhaps because it's designed around proportions).

My use case is that the first column in the `Grid` in my app is a menu column (table) with rows of text, and it looks ugly if it gets stretched too wide since there isn't enough text to fill the empty horizontal space.

**EDIT**: I ended up solving this using css flex box, but if it's possible to do it inside of `chakra-ui`, that'd be good to know.

**Comment by sam-ebe** [comment, marked as answer]
Hi @somguyth ,
I've searched up the documentation, and I didn't find any "clean" solution with the library.
If you use the "maxW" here, you'll end up with a huge gap between the left and the right part on wider screens.
You may try to set the left menu width to a constant value. You may as well go for the full **Flex** solution, with a few calculations on the fly to set the left pannel width when the maxW has been reached (calculation part not included in the following sandbox).

https://codesandbox.io/s/pedantic-wiles-zjt3h?file=/src/App.js
